PICARD-34: AcoustID fingerprint indicator

phw opened the pull request:
<!--
    Hello! Thanks for submitting a pull request to MusicBrainz Picard. We
    appreciate your time and interest in helping our project!

    Use this template to help us review your change. Not everything is required,
    depending on your change. Keep or delete what is relevant for your change.
    Remember that it helps us review if you give more helpful info for us to
    understand your change.

    Ensure that you've read through and followed the Contributing Guidelines, in
    [CONTRIBUTING.md](https://github.com/metabrainz/picard/blob/master/CONTRIBUTING.md).
-->

# Summary
Add a column with a fingerprint indicator.
<!--
    Update the checkbox with an [x] for the type of contribution you are making.
-->

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem
If the fingerprint has been calculated for a file or whether is is ready for submission or already submitted is currently very intransparent in Picard. This became especially obvious after implementation of manual fingerprint generation (PICARD-991).
<!--
    Anything that helps us understand why you are making this change goes here.
    What problem are you trying to fix? What does this change address?
-->

* JIRA ticket (_optional_): PICARD-34
<!--
    Please make sure you prefix your pull request title with 'PICARD-XXX' in order
    for our ticket tracker to link your pull request to the relevant ticket.
-->



# Solution
<!--
    The details of your change. Talk about technical details, considerations, or
    other interesting points. If you have a lot to say, be more detailed in this
    section.
-->
Add a new column (for now hidden by default) which shows the status of fingerprinting with a fingerprint item + tooltip. Possible values are:

- No fingerprint (no icon)
- Submitted fingerprint (gray icon)
- Fingerpint submittable (red icon)

![grafik](https://user-images.githubusercontent.com/29852/72224101-d2f66e80-3576-11ea-9ba1-b062bd8b8923.png)


# Action

<!--
    Other than merging your change, do you want / need us to do anything else
    with your change? This could include reviewing a specific part of your PR.
-->

phw commented:
The macOS build failing is unrelated to this PR. This suddenly happens without change from our side, see PICARD-1705. Need to figure out what changed on the build setup since the last successful build.

zas commented on picard/ui/itemviews.py:
```diff
@@ -256,6 +262,35 @@ def remove(self, objects):
+def paint_fingerprint_icon(painter, rect, icon):
+    if not icon:
+        return
+    size = 16
```
Perhaps use a constant for this, it may be used elsewhere (?)

phw commented:
I fixed the builds and added a constant for the icon size as suggested. Before merging I need to test this properly on macOS if it works for the layout.

zas commented on picard/acoustid/manager.py:
```diff
@@ -62,29 +62,36 @@ def remove(self, file):
             N_('Submitting AcoustIDs ...'),
             echo=None
         )
+        fingerprints = [fingerprint for _, fingerprint in submissions]
```
Even though it is a standard way to declare an unused variable in Python, since it is also used for gettext stuff, I would not use `_` at all.

phw replied:
That's true. It is not really an issue here since this is scoped inside the list comprehension, but you are right we should avoid this.

On a similar note this specific source file makes heavy use of variables named `file`, overwriting the Python `file`. In most places I think we use `file_` for this, but we have a couple of more occurrences of just `file` I think. I will address both in a separate commit.

zas replied:
Yup, the `file` one is really boring, Python guys chose a bad name for their module...
